src/mongocxx/private/index_view.hh

```diff
@@ -134,6 +133,25 @@ class index_view::impl {
             opts_doc.append(bsoncxx::builder::concatenate_doc{session->_get_impl().to_document()});
         }
 
+        if (options.commit_quorum()) {
+            auto server_description = libmongoc::client_select_server(
+                _client, true /* for_writes */, nullptr /* read_prefs */, &error);
+            auto is_master = libmongoc::server_description_ismaster(server_description);
+
+            bson_iter_t iter;
+            bson_iter_init_find(&iter, is_master, "maxWireVersion");
+            int32_t max_wire_version = bson_iter_int32(&iter);
+
+            if (max_wire_version < 9) {
+                throw logic_error{
```

Comment:
Looking at the documentation around the exception classes, `logic_error` is for API misuse. Since this is not an error from a server. Perhaps, in the absence of a more apt error class `write_exception` is more applicable?

Reply:
Done